pkgs/by-name/ed/edopro/package.nix

```
@@ -0,0 +1,267 @@
+        --no-core \
+        --pics=\"${pics_url}\" \
+        --prebuilt-core="${ocgcore}/bin" \
+        --sound=sfml
+    '';
+
+    enableParallelBuilding = true;
+    CXXFLAGS = "-include cstdint";
+    LDFLAGS = "-I ${irrlicht-edopro}/include -L ${irrlicht-edopro}/bin";
+    buildFlags = "verbose=true config=release_x64 ygoprodll";
+    makeFlags = "-C build";
+
+    installPhase = ''
+      mkdir -p $out/bin
+      cp bin/x64/release/ygoprodll $out/bin
+    '';
+  };
+
+  edopro-script =
+    let
+      assetsToCopy = lib.concatStringsSep "," [
+        # Needed if we download files from ProjectIgnis' website or any https only website.
+        "cacert.pem"
+        "config"
+        "deck"
+        "COPYING.txt"
+        "expansions"
+        "lflists"
+        "notices"
+        "puzzles"
+        "fonts"
+        "script"
+        "skin"
+        "sound"
+        "textures"
+        "WindBot"
+      ];
+    in
+    writeShellScriptBin "EDOPro" ''
+      set -eu
+      EDOPRO_DIR="''${XDG_DATA_HOME:-$HOME/.local/share}/edopro"
+
+      if [ ! -d $EDOPRO_DIR ]; then
+          mkdir -p $EDOPRO_DIR
+          cp -r ${assets.src}/{${assetsToCopy}} $EDOPRO_DIR
+
+          find $EDOPRO_DIR -type d -exec chmod u=rwx,go-rwx {} +
+          find $EDOPRO_DIR -type f -exec chmod u=rw,go-rwx {} +
+
+          rm $EDOPRO_DIR/config/io.github.edo9300.EDOPro.desktop.in
+      fi
+
+      export PATH=PATH:'${lib.makeBinPath [ mono ]}';
+      export LD_LIBRARY_PATH='${
+        lib.makeLibraryPath [
+          libGL
+          libX11
+          libxkbcommon
+          libXxf86vm
+          sqlite
+          wayland
+          egl-wayland
+        ]
+      }';
+
+      exec ${edopro}/bin/ygoprodll -C $EDOPRO_DIR $@
+    '';
+
+  edopro-desktop = runCommandLocal "io.github.edo9300.EDOPro.desktop" { } ''
+    cp ${assets.src}/config/io.github.edo9300.EDOPro.desktop.in desktop-template
+
+    sed '/Path=/d' -i desktop-template
+    sed 's/Exec=.*/Exec=EDOPro/' -i desktop-template
+    sed 's/Icon=.*/Icon=EDOPro/' -i desktop-template
+
+    install -D desktop-template $out/share/applications/io.github.edo9300.EDOPro.desktop
+  '';
+in
+symlinkJoin {
+  name = "edopro-application-${deps.edopro-version}";
+  version = deps.edopro-version;
+  paths = [
+    edopro
+    edopro-script
+    edopro-desktop
+  ];
+
+  postBuild = ''
+    mkdir -p $out/share/icons/hicolor/256x256/apps/
+    ${imagemagick}/bin/magick \
+        ${assets.src}/textures/AppIcon.png \
+        -resize 256x256 \
+        $out/share/icons/hicolor/256x256/apps/EDOPro.png
+  '';
+
+  passthru.updateScript = ./update.py;
+
+  meta = with lib; {
+    description = "The bleeding-edge automatic duel simulator, a fork of the YGOPro client.";
+    homepage = "https://projectignis.github.io";
+    changelog = "https://github.com/edo9300/edopro/releases";
+    license = licenses.agpl3Plus;
+    maintainers = with maintainers; [
+      redhawk
+    ];
+    mainprogram = "edopro";
```

https://nixos.org/manual/nixpkgs/unstable/#var-meta-mainProgram

```suggestion
    mainProgram = "edopro";
```